src/gui/render.cpp

```diff
@@ -527,364 +542,144 @@ void RENDER_SetSize(const uint16_t width, const uint16_t height,
-		return;
-	}
-
-	shader_path.replace_extension("");
-
-	const auto it = legacy_name_mappings.find(shader_path.string());
-	if (it != legacy_name_mappings.end()) {
-		const auto new_name = it->second;
-		LOG_WARNING("RENDER: Built-in shader '%s' has been renamed; please use '%s' instead.",
-		            name.c_str(),
-		            new_name.c_str());
-	}
+	const auto shader_info = get_shader_manager().GetCurrentShaderInfo();
+	return shader_info.settings.use_srgb_framebuffer;
 }
-#endif
 
-// Returns true if the shader has been changed, or if a shader has been loaded
-// for the first time since launch
-static bool init_shader([[maybe_unused]] Section* sec)
+static bool is_using_opengl_output_mode()
 {
-#if C_OPENGL
 	assert(control);
-	const Section* sdl_sec = control->GetSection("sdl");
+
+	const auto sdl_sec = static_cast<const Section_prop*>(
+	        control->GetSection("sdl"));
 	assert(sdl_sec);
 
 	const bool using_opengl = starts_with(sdl_sec->GetPropValue("output"),
 	                                      "opengl");
 
-	const auto render_sec = static_cast<const Section_prop*>(
-	        control->GetSection("render"));
-
-	assert(render_sec);
-	auto sh       = render_sec->Get_path("glshader");
-	auto filename = std::string(sh->GetValue());
-
-	constexpr auto fallback_shader = "none";
-	if (filename.empty()) {
-		filename = fallback_shader;
-	} else if (filename == "default") {
-		filename = "interpolation/sharp";
-	}
-
-	if (render.shader.filename == filename) {
-		return false;
-	}
-
-	log_warning_if_legacy_shader_name(filename);
-
-	std::string source = {};
-
-	if (!read_shader_source(sh->realpath.string(), source) &&
-	    (sh->realpath == filename || !read_shader_source(filename, source))) {
-		sh->SetValue("none");
-		source.clear();
-
-		// List all the existing shaders for the user
-		LOG_ERR("RENDER: Shader file '%s' not found", filename.c_str());
-		for (const auto& line : RENDER_InventoryShaders()) {
-			LOG_WARNING("RENDER: %s", line.c_str());
-		}
-
-		// Fallback to the 'none' shader and otherwise fail
-		if (read_shader_source(fallback_shader, source)) {
-			filename = fallback_shader;
-		} else {
-			E_Exit("RENDER: Fallback shader file '%s' not found and is mandatory",
-			       fallback_shader);
-		}
-	}
-
-	// Reset shader settings to defaults
-	render.shader.settings = {};
-
-	if (using_opengl && source.length() && render.shader.filename != filename) {
-		LOG_MSG("RENDER: Using GLSL shader '%s'", filename.c_str());
-		set_shader_settings(source);
-
-		// Move the temporary filename and source into the memebers
-		render.shader.filename = std::move(filename);
-		render.shader.source   = std::move(source);
-
-		// Pass the shader source up to the GFX engine
-		GFX_SetShader(render.shader.source);
-	}
-
-	return true;
-#endif
+	return using_opengl;
 }
 
-void RENDER_InitShader(Section* sec)
+std::deque<std::string> RENDER_InventoryShaders()
 {
-	init_shader(sec);
+	return get_shader_manager().InventoryShaders();
```

Comment:
Maybe `RENDER_GetShadersNames()`? (is that what it's returning?)

`InventoryShaders()` tells me it's doing the inventorying work, and a return value might not be needed. "CollateTheBooks", "InventoryTheStuff"... just do some internal book keeping.

Reply:
Yep, I agree that's better; I just kept the old name. I think it was you who come up with this name, btw 😄

Reply:
Perfect; thanks @johnnovak. It's been a while so I forgot I wrote that. The function name is much clearer now, thanks for that! :rocket: